js/viz/gauges/bar_gauge.js

```diff
@@ -291,27 +291,36 @@ export const dxBarGauge = BaseGauge.inherit({
             const startAngle = bar._bar.attr('startAngle');
             const endAngle = bar._bar.attr('endAngle');
             const coordStart = getStartCoordsArc.apply(null, normalizeArcParams(x, y, innerRadius, outerRadius, startAngle, endAngle));
-            const xStart = Number(coordStart.x);
-            const yStart = Number(coordStart.y);
+            const { cos, sin } = _getCosAndSin(bar._angle);
+            const xStart = coordStart.x - (sin * connectorWidth / 2) - cos;
+            const yStart = coordStart.y - (cos * connectorWidth / 2) + sin;
             const box = bar._text.getBBox();
-            const shiftConnector = connectorWidth / 2;
             const lastCoords = bar._text._lastCoords;
+            const originalPoints = [
+                xStart,
+                yStart,
+                box.x + lastCoords.x,
+                box.y + box.height / 2 + lastCoords.y];
 
             if(bar._angle > 90) {
-                bar._line.attr({ points: [
-                    xStart + shiftConnector,
-                    yStart + shiftConnector,
-                    box.x + box.width + lastCoords.x,
-                    box.y + box.height / 2 + lastCoords.y
-                ] });
-            } else if(bar._angle <= 90) {
-                bar._line.attr({ points: [
-                    xStart - shiftConnector,
-                    yStart + shiftConnector,
-                    box.x + lastCoords.x,
-                    box.y + box.height / 2 + lastCoords.y
-                ] });
+                originalPoints[2] += box.width;
             }
+
+            if(connectorWidth % 2) {
+                const xDeviationForOddConnectorWidth = -sin / 2;
```

Comment:
We see that this deviation of for an odd connector width because this code is placed in `if(connectorWidth % 2)` condition. It's not necessary to copypaste it to the name. Let's simplify it
```suggestion
                const xDeviation = -sin / 2;
```